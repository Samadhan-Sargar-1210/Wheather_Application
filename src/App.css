/* Import Google Fonts for better typography */
@import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700;800;900&display=swap');

/* App.css - Main application styles */
.App {
  width: 100%;
  min-height: 100vh;
  overflow-x: hidden;
}

/* Global styles */
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  font-family: 'Inter', 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  line-height: 1.6;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

/* Smooth scrolling */
html {
  scroll-behavior: smooth;
}

/* Focus styles for accessibility */
*:focus {
  outline: 2px solid #667eea;
  outline-offset: 2px;
}

/* Selection styles */
::selection {
  background: rgba(102, 126, 234, 0.3);
  color: white;
}

/* Scrollbar styling */
::-webkit-scrollbar {
  width: 8px;
}

::-webkit-scrollbar-track {
  background: rgba(255, 255, 255, 0.1);
}

::-webkit-scrollbar-thumb {
  background: rgba(255, 255, 255, 0.3);
  border-radius: 4px;
}

::-webkit-scrollbar-thumb:hover {
  background: rgba(255, 255, 255, 0.5);
}

/* Reduced motion for accessibility */
@media (prefers-reduced-motion: reduce) {
  *,
  *::before,
  *::after {
    animation-duration: 0.01ms !important;
    animation-iteration-count: 1 !important;
    transition-duration: 0.01ms !important;
    scroll-behavior: auto !important;
  }
}

/* High contrast mode support */
@media (prefers-contrast: high) {
  .weather-app {
    border: 2px solid white;
  }
  
  .weather-card {
    border: 2px solid rgba(255, 255, 255, 0.5);
  }
}

/* Print styles */
@media print {
  .weather-app {
    background: white !important;
    color: black !important;
  }
  
  .weather-card {
    background: white !important;
    border: 1px solid #ccc !important;
    box-shadow: none !important;
  }
}

/* Premium Weather App - Enhanced App Styles */

#root {
  width: 100%;
  min-height: 100vh;
  margin: 0;
  padding: 0;
  background: var(--bg-primary);
  transition: background-color var(--transition-smooth);
  overflow-x: hidden;
}

/* Premium App Container */
.app {
  min-height: 100vh;
  width: 100%;
  position: relative;
  overflow-x: hidden;
  background: var(--bg-primary);
  transition: all var(--transition-smooth);
}

/* Premium Loading Screen */
.app-loading {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  background: linear-gradient(135deg, var(--accent-primary), var(--accent-secondary));
  color: white;
  position: relative;
  overflow: hidden;
}

.app-loading::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: 
    radial-gradient(circle at 20% 20%, rgba(255, 255, 255, 0.2) 0%, transparent 50%),
    radial-gradient(circle at 80% 80%, rgba(255, 255, 255, 0.1) 0%, transparent 60%);
  animation: float 8s ease-in-out infinite;
}

.loading-logo {
  font-size: 4rem;
  margin-bottom: 30px;
  animation: logo-bounce 2s ease-in-out infinite;
  filter: drop-shadow(0 8px 16px rgba(0, 0, 0, 0.3));
  z-index: 2;
  position: relative;
}

.loading-text {
  font-size: 1.5rem;
  font-weight: 600;
  margin-bottom: 20px;
  z-index: 2;
  position: relative;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
}

.loading-spinner {
  width: 60px;
  height: 60px;
  border: 4px solid rgba(255, 255, 255, 0.3);
  border-top: 4px solid white;
  border-radius: 50%;
  animation: rotate 1s linear infinite;
  z-index: 2;
  position: relative;
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.2);
}

/* Premium Error Screen */
.app-error {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
  background: linear-gradient(135deg, #ff6b6b, #ffb86b);
  color: white;
  text-align: center;
  padding: 40px;
  position: relative;
  overflow: hidden;
}

.app-error::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: 
    radial-gradient(circle at 30% 30%, rgba(255, 255, 255, 0.1) 0%, transparent 50%),
    radial-gradient(circle at 70% 70%, rgba(255, 255, 255, 0.05) 0%, transparent 60%);
  animation: pulse 4s ease-in-out infinite;
}

.error-icon {
  font-size: 6rem;
  margin-bottom: 30px;
  animation: error-shake 0.5s ease-in-out infinite;
  filter: drop-shadow(0 8px 16px rgba(0, 0, 0, 0.3));
  z-index: 2;
  position: relative;
}

.error-title {
  font-size: 2.5rem;
  font-weight: 700;
  margin-bottom: 20px;
  z-index: 2;
  position: relative;
  text-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
}

.error-message {
  font-size: 1.2rem;
  margin-bottom: 30px;
  max-width: 500px;
  line-height: 1.6;
  z-index: 2;
  position: relative;
  text-shadow: 0 1px 2px rgba(0, 0, 0, 0.3);
}

.error-button {
  background: rgba(255, 255, 255, 0.2);
  backdrop-filter: blur(10px);
  border: 2px solid rgba(255, 255, 255, 0.3);
  border-radius: 12px;
  padding: 12px 24px;
  color: white;
  font-size: 1rem;
  font-weight: 600;
  cursor: pointer;
  transition: all var(--transition-smooth);
  z-index: 2;
  position: relative;
  box-shadow: 0 4px 15px rgba(0, 0, 0, 0.2);
}

.error-button:hover {
  background: rgba(255, 255, 255, 0.3);
  border-color: rgba(255, 255, 255, 0.5);
  transform: translateY(-2px);
  box-shadow: 0 6px 20px rgba(0, 0, 0, 0.3);
}

/* Premium Animation Keyframes */
@keyframes logo-bounce {
  0%, 20%, 50%, 80%, 100% {
    transform: translateY(0);
  }
  40% {
    transform: translateY(-20px);
  }
  60% {
    transform: translateY(-10px);
  }
}

@keyframes error-shake {
  0%, 100% {
    transform: translateX(0);
  }
  10%, 30%, 50%, 70%, 90% {
    transform: translateX(-5px);
  }
  20%, 40%, 60%, 80% {
    transform: translateX(5px);
  }
}

@keyframes float {
  0%, 100% {
    transform: translateY(0px);
  }
  50% {
    transform: translateY(-20px);
  }
}

@keyframes rotate {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@keyframes pulse {
  0%, 100% {
    opacity: 0.5;
  }
  50% {
    opacity: 1;
  }
}

/* Premium Page Transitions */
.page-enter {
  opacity: 0;
  transform: translateY(30px);
}

.page-enter-active {
  opacity: 1;
  transform: translateY(0);
  transition: all var(--transition-smooth);
}

.page-exit {
  opacity: 1;
  transform: translateY(0);
}

.page-exit-active {
  opacity: 0;
  transform: translateY(-30px);
  transition: all var(--transition-smooth);
}

/* Premium Scrollbar Styling */
::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}

::-webkit-scrollbar-track {
  background: var(--bg-secondary);
  border-radius: 4px;
}

::-webkit-scrollbar-thumb {
  background: linear-gradient(135deg, var(--accent-primary), var(--accent-secondary));
  border-radius: 4px;
  transition: all var(--transition-smooth);
}

::-webkit-scrollbar-thumb:hover {
  background: linear-gradient(135deg, var(--accent-secondary), var(--accent-primary));
}

/* Firefox scrollbar */
* {
  scrollbar-width: thin;
  scrollbar-color: var(--accent-primary) var(--bg-secondary);
}

/* Premium Focus Styles */
*:focus {
  outline: 2px solid var(--accent-primary);
  outline-offset: 2px;
  border-radius: 4px;
}

[data-high-contrast="true"] *:focus {
  outline: 3px solid #ffffff;
  outline-offset: 3px;
}

/* Premium Selection Styles */
::selection {
  background: var(--accent-primary);
  color: white;
}

::-moz-selection {
  background: var(--accent-primary);
  color: white;
}

/* Enhanced Responsive Design */
@media (max-width: 768px) {
  .loading-logo {
    font-size: 3rem;
  }
  
  .loading-text {
    font-size: 1.3rem;
  }
  
  .loading-spinner {
    width: 50px;
    height: 50px;
  }
  
  .error-icon {
    font-size: 4rem;
  }
  
  .error-title {
    font-size: 2rem;
  }
  
  .error-message {
    font-size: 1.1rem;
  }
}

@media (max-width: 480px) {
  .loading-logo {
    font-size: 2.5rem;
  }
  
  .loading-text {
    font-size: 1.1rem;
  }
  
  .loading-spinner {
    width: 40px;
    height: 40px;
  }
  
  .error-icon {
    font-size: 3rem;
  }
  
  .error-title {
    font-size: 1.5rem;
  }
  
  .error-message {
    font-size: 1rem;
  }
  
  .error-button {
    padding: 10px 20px;
    font-size: 0.9rem;
  }
}

/* Landscape orientation support */
@media (orientation: landscape) and (max-height: 600px) {
  .app-loading,
  .app-error {
    min-height: 100vh;
    padding: 20px;
  }
  
  .loading-logo {
    font-size: 2.5rem;
    margin-bottom: 20px;
  }
  
  .loading-text {
    font-size: 1.1rem;
    margin-bottom: 15px;
  }
  
  .error-icon {
    font-size: 3rem;
    margin-bottom: 20px;
  }
  
  .error-title {
    font-size: 1.5rem;
    margin-bottom: 15px;
  }
  
  .error-message {
    font-size: 1rem;
    margin-bottom: 20px;
  }
}

/* High contrast mode */
@media (prefers-contrast: high) {
  .app-loading,
  .app-error {
    background: var(--bg-primary);
    color: var(--text-primary);
  }
  
  .loading-spinner {
    border-color: var(--text-primary);
    border-top-color: var(--accent-primary);
  }
  
  .error-button {
    background: var(--accent-primary);
    border-color: var(--accent-primary);
    color: white;
  }
}

/* Reduced motion support */
@media (prefers-reduced-motion: reduce) {
  .app-loading::before,
  .app-error::before,
  .loading-logo,
  .loading-spinner,
  .error-icon {
    animation: none;
  }
  
  .page-enter,
  .page-enter-active,
  .page-exit,
  .page-exit-active {
    transition: none;
  }
}

/* Touch device optimizations */
@media (hover: none) and (pointer: coarse) {
  .error-button:hover {
    transform: none;
  }
  
  .error-button:active {
    transform: scale(0.95);
  }
}

/* Performance optimizations */
.app {
  will-change: background-color;
  backface-visibility: hidden;
  transform: translateZ(0);
}

/* Accessibility improvements */
.app[aria-busy="true"] {
  cursor: wait;
}

.app[aria-busy="true"] * {
  pointer-events: none;
}

/* Print styles */
@media print {
  .app {
    background: white !important;
    color: black !important;
  }
  
  .app-loading,
  .app-error {
    display: none !important;
  }
}

/* Premium Loading States */
.app.loading {
  pointer-events: none;
}

.app.loading::after {
  content: '';
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.1);
  backdrop-filter: blur(2px);
  z-index: 9999;
  pointer-events: none;
}

/* Premium Error States */
.app.error {
  background: linear-gradient(135deg, rgba(255, 107, 107, 0.1), rgba(255, 184, 107, 0.1));
}

/* Premium Success States */
.app.success {
  background: linear-gradient(135deg, rgba(76, 175, 80, 0.1), rgba(139, 195, 74, 0.1));
}

/* Premium Theme Transitions */
.app.theme-transitioning {
  transition: background-color var(--transition-slow);
}

/* Premium Focus Management */
.app:focus-within {
  outline: none;
}

/* Premium Keyboard Navigation */
.app[data-keyboard-navigation="true"] *:focus {
  outline: 3px solid var(--accent-primary);
  outline-offset: 3px;
}

/* Premium Screen Reader Support */
.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border: 0;
}

/* Premium Loading Shimmer */
.shimmer {
  background: linear-gradient(90deg, 
    var(--bg-secondary) 25%, 
    rgba(255, 255, 255, 0.1) 50%, 
    var(--bg-secondary) 75%);
  background-size: 200px 100%;
  animation: shimmer 1.5s infinite;
}

@keyframes shimmer {
  0% {
    background-position: -200px 0;
  }
  100% {
    background-position: calc(200px + 100%) 0;
  }
} 