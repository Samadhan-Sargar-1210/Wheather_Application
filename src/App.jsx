import React from 'react'
import './App.css'
import WeatherApp from './components/WeatherApp'
import ErrorBoundary from './components/ErrorBoundary'
import { ThemeProvider } from './context/ThemeContext'
import ThemeToggle from './components/ThemeToggle'

function App() {
  return (
    <ErrorBoundary>
      <ThemeProvider>
        <div className="App">
          <ThemeToggle />
          <WeatherApp />
        </div>
      </ThemeProvider>
    </ErrorBoundary>
  )
}

export default App 