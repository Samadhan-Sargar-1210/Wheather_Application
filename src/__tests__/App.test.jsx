import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';

// Mock the WeatherApp component
jest.mock('../components/WeatherApp', () => {
  return function MockWeatherApp() {
    return <div data-testid="weather-app">Weather App Component</div>;
  };
});

// Mock i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key) => key,
    i18n: {
      changeLanguage: jest.fn(),
      language: 'en',
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: jest.fn(),
  },
}));

describe('App Component', () => {
  describe('Initial Render', () => {
    test('renders without crashing', () => {
      render(<App />);
      
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });

    test('renders WeatherApp component', () => {
      render(<App />);
      
      expect(screen.getByText('Weather App Component')).toBeInTheDocument();
    });

    test('has correct CSS classes', () => {
      render(<App />);
      
      const appContainer = screen.getByTestId('weather-app').parentElement;
      expect(appContainer).toHaveClass('App');
    });
  });

  describe('Snapshot Tests', () => {
    test('matches snapshot', () => {
      const { container } = render(<App />);
      expect(container).toMatchSnapshot();
    });

    test('matches snapshot with different themes', () => {
      // Test light theme
      const { container: lightContainer } = render(<App />);
      expect(lightContainer).toMatchSnapshot('light-theme');
    });

    test('matches snapshot with different languages', () => {
      // Mock different language
      jest.doMock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key) => key,
          i18n: {
            changeLanguage: jest.fn(),
            language: 'hi',
          },
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

      const { container } = render(<App />);
      expect(container).toMatchSnapshot('hindi-language');
    });
  });

  describe('Integration Tests', () => {
    test('integrates with i18next correctly', () => {
      render(<App />);
      
      // App should render without i18next errors
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });

    test('handles theme context integration', () => {
      render(<App />);
      
      // App should render with theme context
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });

    test('handles error boundaries', () => {
      // Mock WeatherApp to throw error
      jest.doMock('../components/WeatherApp', () => {
        return function MockWeatherApp() {
          throw new Error('Test error');
        };
      });

      // Should not crash the app
      expect(() => {
        render(<App />);
      }).not.toThrow();
    });
  });

  describe('Performance Tests', () => {
    test('renders efficiently', () => {
      const startTime = performance.now();
      render(<App />);
      const endTime = performance.now();
      
      expect(endTime - startTime).toBeLessThan(50); // Should render in less than 50ms
    });

    test('does not cause memory leaks', () => {
      const { unmount } = render(<App />);
      
      // Unmount should not cause issues
      expect(() => {
        unmount();
      }).not.toThrow();
    });
  });

  describe('Accessibility Tests', () => {
    test('has proper document title', () => {
      render(<App />);
      
      expect(document.title).toBe('Weather App - Get Real-time Weather Updates');
    });

    test('has proper meta description', () => {
      render(<App />);
      
      const metaDescription = document.querySelector('meta[name="description"]');
      expect(metaDescription).toHaveAttribute('content', expect.stringContaining('weather'));
    });

    test('has proper viewport meta tag', () => {
      render(<App />);
      
      const viewportMeta = document.querySelector('meta[name="viewport"]');
      expect(viewportMeta).toHaveAttribute('content', expect.stringContaining('width=device-width'));
    });
  });

  describe('Error Handling', () => {
    test('handles missing dependencies gracefully', () => {
      // Mock missing dependency
      const originalConsoleError = console.error;
      console.error = jest.fn();
      
      // Should not crash when dependencies are missing
      expect(() => {
        render(<App />);
      }).not.toThrow();
      
      console.error = originalConsoleError;
    });

    test('handles initialization errors gracefully', () => {
      // Mock i18next initialization error
      jest.doMock('react-i18next', () => ({
        useTranslation: () => {
          throw new Error('i18next initialization error');
        },
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

      // Should not crash on initialization errors
      expect(() => {
        render(<App />);
      }).not.toThrow();
    });
  });

  describe('Browser Compatibility', () => {
    test('works in different browsers', () => {
      // Mock different user agents
      const originalUserAgent = navigator.userAgent;
      
      // Chrome
      Object.defineProperty(navigator, 'userAgent', {
        value: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36',
        configurable: true,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
      
      // Firefox
      Object.defineProperty(navigator, 'userAgent', {
        value: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:89.0) Gecko/20100101 Firefox/89.0',
        configurable: true,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
      
      // Safari
      Object.defineProperty(navigator, 'userAgent', {
        value: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1.1 Safari/605.1.15',
        configurable: true,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
      
      // Restore original user agent
      Object.defineProperty(navigator, 'userAgent', {
        value: originalUserAgent,
        configurable: true,
      });
    });
  });

  describe('SEO Tests', () => {
    test('has proper Open Graph meta tags', () => {
      render(<App />);
      
      const ogTitle = document.querySelector('meta[property="og:title"]');
      const ogDescription = document.querySelector('meta[property="og:description"]');
      const ogType = document.querySelector('meta[property="og:type"]');
      
      expect(ogTitle).toHaveAttribute('content', expect.stringContaining('Weather App'));
      expect(ogDescription).toHaveAttribute('content', expect.stringContaining('weather'));
      expect(ogType).toHaveAttribute('content', 'website');
    });

    test('has proper Twitter Card meta tags', () => {
      render(<App />);
      
      const twitterCard = document.querySelector('meta[name="twitter:card"]');
      const twitterTitle = document.querySelector('meta[name="twitter:title"]');
      const twitterDescription = document.querySelector('meta[name="twitter:description"]');
      
      expect(twitterCard).toHaveAttribute('content', 'summary_large_image');
      expect(twitterTitle).toHaveAttribute('content', expect.stringContaining('Weather App'));
      expect(twitterDescription).toHaveAttribute('content', expect.stringContaining('weather'));
    });

    test('has proper keywords meta tag', () => {
      render(<App />);
      
      const keywordsMeta = document.querySelector('meta[name="keywords"]');
      expect(keywordsMeta).toHaveAttribute('content', expect.stringContaining('weather'));
    });
  });

  describe('Responsive Design Tests', () => {
    test('adapts to different screen sizes', () => {
      // Test mobile viewport
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 375,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
      
      // Test tablet viewport
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 768,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
      
      // Test desktop viewport
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 1024,
      });
      
      render(<App />);
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });
  });

  describe('Security Tests', () => {
    test('has proper CSP meta tag', () => {
      render(<App />);
      
      const cspMeta = document.querySelector('meta[http-equiv="Content-Security-Policy"]');
      if (cspMeta) {
        expect(cspMeta).toHaveAttribute('content');
      }
    });

    test('has proper X-Frame-Options', () => {
      render(<App />);
      
      // This would typically be set by the server, but we can test if the app
      // handles it properly when present
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });
  });

  describe('Loading States', () => {
    test('handles loading states gracefully', () => {
      render(<App />);
      
      // App should render immediately without loading states
      expect(screen.getByTestId('weather-app')).toBeInTheDocument();
    });
  });

  describe('Memory Management', () => {
    test('cleans up resources on unmount', () => {
      const { unmount } = render(<App />);
      
      // Should not cause memory leaks
      expect(() => {
        unmount();
      }).not.toThrow();
    });

    test('handles multiple mount/unmount cycles', () => {
      for (let i = 0; i < 10; i++) {
        const { unmount } = render(<App />);
        expect(screen.getByTestId('weather-app')).toBeInTheDocument();
        unmount();
      }
    });
  });
}); 