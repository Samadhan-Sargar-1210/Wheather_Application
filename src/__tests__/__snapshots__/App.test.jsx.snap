// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot Tests matches snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      aria-label="weather.theme.selector"
      class="theme-toggle"
      data-testid="theme-toggle"
      role="group"
    >
      <button
        aria-label="weather.theme.light"
        aria-pressed="true"
        class="theme-button light active"
      >
        <span
          class="theme-icon"
        >
          ☀️
        </span>
        <span
          class="theme-label"
        >
          Light
        </span>
      </button>
      <button
        aria-label="weather.theme.dark"
        aria-pressed="false"
        class="theme-button dark "
      >
        <span
          class="theme-icon"
        >
          🌙
        </span>
        <span
          class="theme-label"
        >
          Dark
        </span>
      </button>
      <button
        aria-label="weather.theme.high-contrast"
        aria-pressed="false"
        class="theme-button high-contrast "
      >
        <span
          class="theme-icon"
        >
          ♿
        </span>
        <span
          class="theme-label"
        >
          High Contrast
        </span>
      </button>
    </div>
    <div
      data-testid="weather-app"
    >
      Weather App Component
    </div>
  </div>
</div>
`;

exports[`App Component Snapshot Tests matches snapshot with different languages: hindi-language 1`] = `
<div>
  <div
    class="App"
  >
    <div
      aria-label="weather.theme.selector"
      class="theme-toggle"
      data-testid="theme-toggle"
      role="group"
    >
      <button
        aria-label="weather.theme.light"
        aria-pressed="true"
        class="theme-button light active"
      >
        <span
          class="theme-icon"
        >
          ☀️
        </span>
        <span
          class="theme-label"
        >
          Light
        </span>
      </button>
      <button
        aria-label="weather.theme.dark"
        aria-pressed="false"
        class="theme-button dark "
      >
        <span
          class="theme-icon"
        >
          🌙
        </span>
        <span
          class="theme-label"
        >
          Dark
        </span>
      </button>
      <button
        aria-label="weather.theme.high-contrast"
        aria-pressed="false"
        class="theme-button high-contrast "
      >
        <span
          class="theme-icon"
        >
          ♿
        </span>
        <span
          class="theme-label"
        >
          High Contrast
        </span>
      </button>
    </div>
    <div
      data-testid="weather-app"
    >
      Weather App Component
    </div>
  </div>
</div>
`;

exports[`App Component Snapshot Tests matches snapshot with different themes: light-theme 1`] = `
<div>
  <div
    class="App"
  >
    <div
      aria-label="weather.theme.selector"
      class="theme-toggle"
      data-testid="theme-toggle"
      role="group"
    >
      <button
        aria-label="weather.theme.light"
        aria-pressed="true"
        class="theme-button light active"
      >
        <span
          class="theme-icon"
        >
          ☀️
        </span>
        <span
          class="theme-label"
        >
          Light
        </span>
      </button>
      <button
        aria-label="weather.theme.dark"
        aria-pressed="false"
        class="theme-button dark "
      >
        <span
          class="theme-icon"
        >
          🌙
        </span>
        <span
          class="theme-label"
        >
          Dark
        </span>
      </button>
      <button
        aria-label="weather.theme.high-contrast"
        aria-pressed="false"
        class="theme-button high-contrast "
      >
        <span
          class="theme-icon"
        >
          ♿
        </span>
        <span
          class="theme-label"
        >
          High Contrast
        </span>
      </button>
    </div>
    <div
      data-testid="weather-app"
    >
      Weather App Component
    </div>
  </div>
</div>
`;
